### PR TITLE
docs(general): create a concepts page for resource authorizations

### PR DESCRIPTION
**Pull request by Ben-Sheppard**
## Description

Closes https://github.com/camunda-cloud/identity/issues/2116

As part of the resource based authorizations in SaaS epic it because clear that there were a couple of documentation pieces missing that would help support a good user experience. This PR introduces the first new page of those tasks which is centred on the concept of what RBA actually is and also how it applies to the components in the stack at the moment.

## When should this change go live?

- [x] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/concepts/resource-authorizations.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - Update process instance
+  - Delete process instance
+- Decision definition:
+  - Read
+  - Delete
+
+#### Tasklist
+
+- Process definition:
+  - Start process instance
```
Comment (akeller) [inline docs/components/concepts/resource-authorizations.md:27]: I like how this page is looking, but structurally (and possibly nitpicky), we should drop the headings/headers here unless we plan on adding more information (as in, more sentences) to these sections.

**Comment by akeller** [comment]
Pieces I'm currently missing in the draft -

* Does this relate to [management of users of your organization](https://docs.camunda.io/docs/components/console/manage-organization/manage-users/)?
  * https://github.com/camunda/camunda-platform-docs/pull/2366 covers some of this
* How granular is this? By user? By role? By the individual point in the list? 

If it's easier to start merging these existing RBA docs PRs so you can work on a single PR against `main`, let's do that.

**Comment by Ben-Sheppard** [comment]
> Pieces I'm currently missing in the draft -
> 
> * Does this relate to [management of users of your organization](https://docs.camunda.io/docs/components/console/manage-organization/manage-users/)?
>   
>   * [feat(comp/console): resource based auth #2366](https://github.com/camunda/camunda-platform-docs/pull/2366) covers some of this
> * How granular is this? By user? By role? By the individual point in the list?
> 
> If it's easier to start merging these existing RBA docs PRs so you can work on a single PR against `main`, let's do that.

It does relate to that page yes, what I would like to do is once the #2366 PR is in, then link to this page from the console page.

Currently this is by user, with the intention in the future of including group support (the self-managed offering already supports groups)

File: docs/components/concepts/resource-authorizations.md (hunk 1 above)
Comment (Ben-Sheppard) [reply docs/components/concepts/resource-authorizations.md:27]: Yea I wasn't too sure about the structure, I wanted to convey the relevancy of each permission with the context of where its applicable so headings felt like a nice way to do that but would you recommend just leaving the headings off and maybe emboldening the component names instead?

Comment (christinaausley) [reply docs/components/concepts/resource-authorizations.md:27]: I can open an issue in my backlog to give a little more love to the Self-Managed landing page, adding descriptions below the components and such, similar to my work for the [SaaS Components page](https://github.com/camunda/camunda-platform-docs/pull/2168). This should help remove the necessity of adding headers.

**Comment by christinaausley** [comment]
THANK YOU for using Camunda 8!

**Comment by akeller** [comment]
> Currently this is by user, with the intention in the future of including group support (the self-managed offering already supports groups)

This is a great example of something I **would** document then - "Group support is currently available in Self-Managed." and link to info on that. That way the docs can be used to help people decide if they should start building on SaaS or SM (or wait for the functionality in SaaS).

**Comment by christinaausley** [comment]
FYI https://github.com/camunda/camunda-platform-docs/pull/2587 -- will assist with any merge conflicts.

File: docs/components/concepts/resource-authorizations.md (hunk 1 above)
Comment (Ben-Sheppard) [reply docs/components/concepts/resource-authorizations.md:27]: Perfect, thank you @christinaausley, for now I will update this PR with my suggestion above to remove the headers for the moment

**Review by christinaausley — APPROVED** [review]

**Review by akeller — APPROVED** [review]
LGTM.

Merging this will give us a change to look at the changes in staging and make sure we are happy with linking across the PRs, identifying any remaining gaps, etc.

**Pull request merged**